core/src/main/java/novoda/lib/sqliteprovider/util/UriUtils.java

```diff
@@ -6,29 +6,28 @@
 
 public class UriUtils {
 
-    private static Map<String, String> mappedIds = new HashMap<String, String>();
+    private UriUtils() throws IllegalAccessException {
+        throw new IllegalAccessException("Utils class shouldn't be instantiated");
+    }
 
-    public static UriUtils from(Uri uri) {
-        UriUtils utils = new UriUtils();
+    public static Map<String, String> mapIds(Uri uri) {
+        Map<String, String> mappedIds = new HashMap<>();
 
         final List<String> segs = uri.getPathSegments();
         String parent = "";
 
         for (int i = 0; i < segs.size(); i++) {
             String currentSeg = segs.get(i);
             if (isNumeric(currentSeg)) {
-                final int children = Integer.parseInt(currentSeg);
-                for (int l = 0; l < children; l++) {
```

Comment:
Sorry, guys, no idea. 

It's also excluding negative values, not just `0` though. So we might introduce some behavioural change here. Probably not an issue though. We might check on the usage side again though to learn what kind of Uris to expect here. Would be great to have that documented.

Disclaimer: I wasn't involved in any development of this library.